packages/date-picker/src/vaadin-date-picker-overlay-content.js

```diff
@@ -309,15 +244,11 @@ class DatePickerOverlayContent extends ControllerMixin(ThemableMixin(DirMixin(Po
    * @private
    */
   get __useSubMonthScrolling() {
-    return this.$.monthScroller.clientHeight < this.$.monthScroller.itemHeight + this.$.monthScroller.bufferOffset;
-  }
-
-  get calendars() {
-    return [...this.shadowRoot.querySelectorAll('vaadin-month-calendar')];
```

Comment:
Changed logic to only get calendars instances once, because in practice they are never re-rendered.